DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h

```diff
@@ -48,53 +51,93 @@ class MCCompLabel
   static constexpr ULong64_t maskFull = (ul0x1 << (nbitsTrackID + nbitsEvID + nbitsSrcID)) - 1;
 
   MCCompLabel(int trackID, int evID = 0, int srcID = 0) { set(trackID, evID, srcID); }
-  MCCompLabel() = default;
+  MCCompLabel(unsigned int trackID, int evID, int srcID, bool fake) { set(trackID, evID, srcID, fake); }
```

Comment:
See below, I'd probably drop the constructor with automatic conversion of negative labels.

The constructor with bool invalid = false is a bit difficult to understand. If invalid = true, we set mLabel explicitly, otherwise we take the C++11 default. Perhaps I'd also add an else case.

Reply:
invalid==false should create a label in NotSet state, added else for clarity